Services/Administration/OJS.Services.Administration.Business/Implementations/ParticipantScoresBusinessService.cs

```diff
@@ -31,6 +34,7 @@ public async Task RecalculateForParticipantByProblem(int participantId, int prob
         if (submission != null)
         {
             await this.participantScoresData.ResetBySubmission(submission);
+            await this.testRunsDataService.DeleteBySubmission(submission.Id);
```

Comment:
Where is this used? Are we covering all the cases? I don't think the best place to clear test runs is in the recalculation of scores, as this recalculation might not be needed for some submissions (but deletion of test runs will be)

Reply:
Yes, it should not be in the RecalculateForParticipant, because this method triggers only if the submission is best submission. The  deletion of testRuns was moved inside the Retest method in SubmissionsBusinessService. In this way it will be always called when submission is retested and since it is in transaction scope, will assure that the test runs will be deleted only if submission is again marked for processing